src/Generator.cpp

```diff
@@ -850,6 +850,15 @@ gengen
             output_types.insert(OutputFileType::registration);
             output_types.insert(OutputFileType::static_library);
         } else {
+            // if emit_flags contains "stmt_viz", add "assembly" to the list
+            for (const auto &f : emit_flags) {
```

Comment:
Hrm... not sure about this. I think it would be preferable to just fail with a message that specifying `stmt_viz` requires that `assembly` also be specified, if for no other reason than it makes it easier to integrate with Blaze/Bazel, which is very picky about undeclared files being generated. (Other Halide contributors may disagree here.)